services/fromage/server.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import os
+import time
+from fromage import models
+from fromage import utils
+import torch
+import sentry_sdk
+from flask import Flask, request, jsonify
+from sentry_sdk.integrations.flask import FlaskIntegration
+
+sentry_sdk.init(dsn=os.getenv("SENTRY_DSN"), integrations=[FlaskIntegration()])
+
+logging.basicConfig(format="%(asctime)s - %(name)s - %(levelname)s - %(message)s", level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+FILE_SERVER_URL = os.getenv("FILE_SERVER_URL")
+RET_SCALE_FACTOR = int(os.getenv("RET_SCALE_FACTOR"))
+
+
+try:
+    model_dir = "/services/fromage/fromage_model"
+    model = models.load_fromage(model_dir)
+
+    if torch.cuda.is_available():
+        logger.info("fromage is set to run on cuda")
+
+    logger.info("fromage is ready")
+except Exception as e:
+    sentry_sdk.capture_exception(e)
+    logger.exception(e)
+    raise e
+
+app = Flask(__name__)
+logging.getLogger("werkzeug").setLevel("WARNING")
+
+
+def generate_responses(image_path, prompt):
+    logger.info(f"prompt generate responses {prompt}")
+    inp_image = [utils.get_image_from_url(image_path)]
+    if prompt == "":
+        prompt = ["What is the image?"]
+
+    logger.info(f"prompts {prompt}")
+    text = ""
+    for p in prompt:
```

Comment:
does this means you may have several prompts in the same utterance? what if the sentence is a string, then you will iterate over letters. or sentence is a list?